pkg/etcdcli/etcdcli.go

```diff
@@ -329,9 +317,63 @@ func (g *etcdClientGetter) UnhealthyMembers() ([]*etcdserverpb.Member, error) {
 		g.eventRecorder.Warningf("UnhealthyEtcdMember", "unhealthy members: %v", strings.Join(unhealthyMemberNames, ","))
 	}
 
+	_, unhealthyMembers := etcdMemberHealth.MemberStatus()
+
 	return unhealthyMembers, nil
 }
 
+func (g *etcdClientGetter) MemberHealth(etcdMembers []*etcdserverpb.Member) (*memberHealth, error) {
+	memberHealth, err := g.getMemberHealth(etcdMembers)
+	if err != nil {
+		return nil, err
+	}
+	return memberHealth, nil
+}
+
+func (g *etcdClientGetter) getMemberHealth(etcdMembers []*etcdserverpb.Member) (*memberHealth, error) {
+	var wg sync.WaitGroup
+	memberHealth := &memberHealth{}
+	hch := make(chan healthCheck, len(etcdMembers))
```

Comment:
(Aside: looks like if there were a thread-safe array construct that would do the same thing but in a more obvious way? Go doesn't give us a lot to work with here... sync.Map would be okay but annoying since we just want an array...)